Healthcare section

cesidio opened the pull request.

Commit:
Create a skeleton rails application

Sets up a scaffold for building on top of. Assumes we will use slim for
html templating, rspec, pry, better errors & postgres for the database.

Commit:
Enable uuid extension in postgres

Ultimately this will allow us to use uuid's for primary keys on our
ActiveRecord models.

Commit:
Introduce an Escort wrapper model with a Detainee

Commit:
Configure default gov styles

This introduces an empty homepage controller for demoing
as well as the cells layer for handling views. This provides
a foundation to build the search from.

This commit introduces the cells gem which provides a way to
encapsulate view logic - templates are accompanied with view
models.

Some of the benefits include not relying on rails helpers which
can pollute the view layer as well as testing edge cases more
easily - no complicated controller specs and less heavy feature
specs.

Commit:
Set explicit path for cell view templates

This is a work around as cell templates cannot be found when
the app is ran in production mode on the docker container.

This can't be reproduced locally running the app in production
mode.

Commit:
Introduce escort search form

Provides validation on the prison number the user is
searching for.

In order for this object to work in form_for scenario
it requires certain behaviours from ActiveModel - these
are documented in the spec.

This introduces Reform as the framework for building out
form objects.

Commit:
Provide basic feedback on validity of search

This commit introduces the search endpoint and connects
the users input and validates it showing errors when
necessary.

Handling results and no result cases will be introduced
next.

This introduces the concept of using assign_attributes/1
as an alias to Reforms validate/1 method which assigns
values to the properties declared in the form object and
then calls valid on the form. Calling valid on the form
is required to set errors however we can't use a direct
call to valid?/invalid? in the view model as an empty form
which is rendered in the show action would be invalid and
therefore show errors when the user has yet to make a
submission.

This commit also introduces some styles to the search box.

Commit:
Search for an Escort by Detainee prison number

We need a way to lookup escorts from the search page
based on prison number.

This is something that can be used in the form to return
results/no results if the prison number is valid.

cesidio committed:
Add Authentication with Devise

We use devise and devise_invitable gems to handle
authentication.

We set devise to use only the following modules:

- invitable
- database_authenticatable
- recoverable
- trackable
- validatable
- timeoutable (timeout set at 60 mins)

Commit:
Allow search form to return an escort when valid

Commit:
Display create action when no results found

This commit just handles the view layer, the actual
call to action doesn't link to an end point yet.

Commit:
Introduce govuk_elements_form_builder gem

This gem helps building forms with a defined style that
is consistent across GovUK projects.

Adding this in now so that Cesidio can work in parallel.

Commit:
Create escort record with prepopulated detainee

Commit:
Extract base form object

Commit:
Introduce a detainee details form object

Commit:
Make cell rendering available to AppCtrller

This is common functionality that will be required across
most controllers.

Commit:
Introduce a detainee details page

This makes use of the DetaineeDetails form object. We
expect that this page/form will be accessible after
clicking the create profile button & for editing from
the profile page(yet to be introduced).

The data will eventually be pulled from a NOMIS api or
copied from a previous escorts detainee record if its
within a certain amount of time(determined by the biz).

Commit:
Make Escort model lookup available to AppCtrller

This is common functionality that will be required across
most controllers.

Commit:
Introduce a temporary profile page

In order to make life easier when testing and demoing in
the browser this seems quite sensible.

Alter the successful save action on detainee details to
redirect to the profile page.

Commit:
Set a default date format for rails to use

We don't want to tell the user to write dates in a
particular format only to throw another format at them
when rendering out saved data.

Commit:
Add a Move model to the Escort

Commit:
Add Move's Destination model

Commit:
Introduce a Move Information form

This form also includes its child form(MoveDestination)
which is used when rendering out objects in the moves
collection.

Commit:
Default the 'from' location in Move Info form

Commit:
Add a move information page

Includes the ability to add multiple destinations without
the use of JS if need be.

Commit:
Include a move information page link on profile

Commit:
Introduce a search view when results are found

Commit:
Introduce rspec-cells & capybara for testing

Commit:
Cells Rails 5 testing workaround

A temporary workaround has been put in place to support
Cells with Rails 5. Please check cells/cells-rails for
updates so that the gem can be upgraded and this can be
removed.

Rails 5 no longer supports creating a TestRequest without
providing args to the initializer however we can imitate the
original behaviour using .create

Commit:
Spec homepage search views

Commit:
Adds a simple feature spec for the happy path

cesidio committed:
Add dev:prime rake task for development env seed

Commit:
Configure CircleCI for automating running of specs

Commit:
Merge pull request #2 from ministryofjustice/configure-circle-ci

Configure CircleCI for automating running of specs

Commit:
Introduce Rubocop with some default styles

Applies said styles to current code base

Commit:
Ensure Rubocop cop runs as part of default rake

cesidio committed:
Merge pull request #3 from ministryofjustice/rubocop

Add Rubocop for code consistency

cesidio committed:
Introduce healthcare model

cesidio committed:
Add physical healthcare section

dan-palmer commented on app/models/forms/healthcare/physical.rb:
```diff
@@ -0,0 +1,15 @@
+module Forms
+  module Healthcare
+    class Physical < Forms::Base
```
Is there a spec incoming for this form?

cesidio committed:
Add mental healthcare form

This is just a form object, we intend to use this to
build a flow.

cesidio committed:
Add social healthcare form

This is just a form object, we intend to use this to
build a flow.

Commit:
Fix position of 'sign out' link

Commit:
Adds a linked list to support building flows

Commit:
Introduce ability to navigate healthcare forms

Commit:
Merge pull request #1 from ministryofjustice/devise

Add Authentication with Devise

cesidio committed:
Introduce remaining healthcare forms

In this commit we introduce:

- allergies
- medical health needs
- transport needs
- medical contact
